code/iexfeed/iex.q

```diff
@@ -0,0 +1,66 @@
+\d .iex
+
+main_url:@[value;`main_url;"https://api.iextrading.com"];
+convert_epoch:@[value;`convert_epoch;{"p"$1970.01.01D+1000000j*x}];
+reqtype:@[value;`reqtype;`both];
+syms:@[value;`syms;`CAT`DOG];
+callback:@[value;`callback;".u.upd"];
+callbackhandle:@[value;`callbackhandle;0i];
+callbackconnection:@[value;`callbackconnection;`];
+quote_suffix:@[value;`quote_suffix;{[sym] "/1.0/stock/",sym,"/quote"}];
+trade_suffix:@[value;`trade_suffix;{[sym]"/1.0/tops/last?symbols=",sym}];
+upd:@[value;`upd;{[t;x] .iex.callbackhandle(.iex.callback;t; value flip x)}];
+timerperiod:@[value;`timerperiod;0D00:00:02.000];
+
+.iex.init:{[x]
+   if[`main_url in key x;.iex.main_url:x[`main_url]];
+   if[`quote_suffix in key x;.iex.quote_suffix:x[`quote_suffix]];
+   if[`trade_suffix in key x;.iex.trade_suffix:x[`trade_suffix]];
+   if[`syms in key x;.iex.syms: upper x[`syms]];
+   if[`reqtype in key x;.iex.reqtype:x[`reqtype]];
+   if[`callbackconnection in key x;.iex.callbackhandle :neg[hopen[.iex.callbackconnection:x[`callbackconnection]]]];
+   if[`callbackhandle in key x;.iex.callbackhandle:x[`callbackhandle]];
+   if[`callback in key x;.iex.callback: $[.iex.callbackhandle=0; string @[value;x[`callback];{[x;y]x set {[t;x]x}}[x[`callback]]]; x[`callback]]];
+   if[`upd in key x; .iex.upd:x[`upd]];
+   .iex.timer:$[not .iex.reqtype in key .iex.timer_dict;'`timer;.iex.timer_dict[.iex.reqtype]];
+   }
```

Comment:
Good stuff getting this changed, but now I feel bad for not being greedier the first time round!
As a general rule I think it is better to take advantage of right to left evaluation to remove brackets. For example with ``x[`main_url]`` the brackets can be removed without harming the functionality. On line 21 we see plenty of brackets that can be reduced to: ``.iex.callbackhandle:neg hopen .iex.callbackconnection:x`callbackconnection``. I'd argue that readability is drastically improved.

An additional point would be to avoid the space in line 21 when setting `.iex.callbackhandle`. At a casual glance it may appear that `:` is being used to terminate early instead of setting a value.